src/modules/lua-support/mtev/HttpClient.lua

```diff
@@ -107,11 +114,41 @@ function HttpClient:do_request(method, uri, _headers, payload, http_version)
       sstr = sstr .. payload
     end
     self.e:write(sstr)
+
+    if self.on_timeout ~= nil then
+      self.request_started_s, self.request_started_us = mtev.gettimeofday()
+    end
```

Comment:
I would set these unconditionally.

Now, if someone does a do_request and sets a timeout in a callback, you'll attempt to put a callback in place with no request start time.  Seems easier to just always record the request start time instead of preventing that (possibly) legitimate use.